### PR TITLE
initial items set-up 

### PR DESCRIPTION
**Pull request by JustasBan**
with db, class and misc. types

### CODE REVIEW

**Comment by The-Blue-K** [comment]
Something strange with github, where after pulling project, most of the projects files say they need to be committed. Most of these changes are auto-generated.

File: kinder-app/Data/Migrations/ApplicationDbContextModelSnapshot.cs [hunk 1]
```diff
@@ -219,6 +219,48 @@ protected override void BuildModel(ModelBuilder modelBuilder)
...
                 });
 
+            modelBuilder.Entity("kinder_app.Models.Item", b =>
+                {
+                    b.Property<int>("ID")
+                        .ValueGeneratedOnAdd()
+                        .HasColumnType("int")
+                        .HasAnnotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn);
+
+                    b.Property<int>("Cathegory")
```
Comment (The-Blue-K) [inline kinder-app/Data/Migrations/ApplicationDbContextModelSnapshot.cs:229]: Category*

File: kinder-app/Models/CathegoryEnum.cs [hunk 2]
```diff
@@ -0,0 +1,10 @@
+﻿namespace kinder_app.Models
+{
+    public enum CathegoryEnum
```
Comment (The-Blue-K) [inline kinder-app/Models/CathegoryEnum.cs:3]: Category*

File: kinder-app/Models/ConditionEnum.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿namespace kinder_app.Models
+{
+    public enum ConditionEnum
+    {
+        Mint,
+        Near_mint,
+        Very_good,
+        Good,
+        Fair
```
Comment (The-Blue-K) [inline kinder-app/Models/ConditionEnum.cs:9]: Might wanna add conditions for poor, very poor and broken.

File: kinder-app/Models/Item.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+            if (other == null) return false;
+            return (this.ID.Equals(other.ID));
+        }
+
+        public int ID { get; set; }
+
+        public DateTime DateOfPurchase { get; set; }
+
+        public ConditionEnum Condition { get; set; }
+        public CathegoryEnum Cathegory { get; set; }
```
Comment (The-Blue-K) [inline kinder-app/Models/Item.cs:57]: Category*

File: kinder-app/Controllers/ItemsController.cs [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+        // POST: Items/Delete/5
+        [HttpPost, ActionName("Delete")]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> DeleteConfirmed(int id)
+        {
+            var item = await _context.Item.FindAsync(id);
+            _context.Item.Remove(item);
+            await _context.SaveChangesAsync();
+            return RedirectToAction(nameof(Index));
+        }
```
Comment (The-Blue-K) [inline kinder-app/Controllers/ItemsController.cs:146]: Why are there no Authorize annotations? now anyone can create, edit and delete items without being logged in.

File: kinder-app/Data/Migrations/20211130215847_InitialItems.cs [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        {
+            migrationBuilder.CreateTable(
+                name: "Item",
+                columns: table => new
+                {
+                    ID = table.Column<int>(type: "int", nullable: false)
+                        .Annotation("SqlServer:Identity", "1, 1"),
+                    DateOfPurchase = table.Column<DateTime>(type: "datetime2", nullable: false),
+                    Condition = table.Column<int>(type: "int", nullable: false),
+                    Cathegory = table.Column<int>(type: "int", nullable: false),
```
Comment (The-Blue-K) [inline kinder-app/Data/Migrations/20211130215847_InitialItems.cs:18]: Should be changed to category in the model so the mistake is not reflected in db.

File: kinder-app/Models/Item.cs [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+        }
+
+        public override string ToString()
+        {
+            string str = "";
+
+            str += ID.ToString() + ';';
+            str += DateOfPurchase.ToString("yyyy-MM-dd") + ';';
+            str += Condition.ToString() + ';';
+            str += Cathegory.ToString() + ';';
```
Comment (The-Blue-K) [inline kinder-app/Models/Item.cs:36]: Category*

**Comment by JustasBan** [comment]
git ignore prob not set up correctly

**Review by The-Blue-K — CHANGES_REQUESTED** [review]
Authorization needs to be added to the creation, editing and deletion of items.
Please correct the repeating misspelling of category as cathegory.

File: kinder-app/Models/ConditionEnum.cs (hunk 3 above)
Comment (JustasBan) [reply kinder-app/Models/ConditionEnum.cs:9]: poor and very poor fall behind fair category but broken is a nice addition

File: kinder-app/Controllers/ItemsController.cs (hunk 5 above)
Comment (JustasBan) [reply kinder-app/Controllers/ItemsController.cs:146]: i will add that to other PR, right now just initialise

**Comment by JustasBan** [comment]
future improvements needed

**Pull request merged**